drivers/scheduler/k8s/k8s.go

```diff
@@ -5113,11 +5134,89 @@ func (k *K8s) CreateCsiSnapsAndValidate(ctx *scheduler.Context, snapClass string
 	return nil
 }
 
-func (k *K8s) createSnapshotAndVerify(data, snapName, namespace, snapClass, restoredPVCName, originalPod, originalPVC string) error {
-	elasticsearchSC := "elasticsearch-sc"
+// CloneFAVolumeAndValidate create new PVC by cloning an existing PVC and make sure the contents of volume are same
+// (Testrail cases C58509)
+func (k *K8s) CloneFAVolumeAndValidate(ctx *scheduler.Context, request scheduler.CSICloneRequest)  error {
+	// This test will validate the content of the snapshot as opposed to just verify creation of snapshot.
+	timestamp := request.Timestamp
+	namespace := request.Namespace + timestamp
+	clientset, err := getKubeClient("")
+	if err != nil {
+		logrus.Errorf("Failed to get kube client for clone test: %s", err)
+		return err
+	}
+
+	storageClassName := request.StorageclassName
+	// for FADA test, find a FA SC on our own that has pure_block param, if no FA SC, skip the test
+	if request.FADA {
+		fadaStorageClassName := ""
+		SCs, err:= clientset.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			logrus.Errorf("Failed to get list of SCs from cluster")
+			return err
+		}
+
+		for _, SC := range SCs.Items {
+			if SC.Parameters["backend"] == PureBlock {
+				fadaStorageClassName = SC.Name
+			}
+		}
+
+		if fadaStorageClassName == "" {
+			logrus.Warn("No FADA SC on the cluster, skipping the test")
+			return nil
+		}
+		storageClassName = fadaStorageClassName
+	}
+	logrus.Infof("Procedding with SC %s", storageClassName)
+
+	// create a namespace for snapshot and restore test
+	nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_, err = k8sCore.CreateNamespace(nsSpec)
```

Comment:
Any reason we don't want to use the existing namespace? Only reason I ask is because I know torpedo adds some special labels to the namespace to clean up, so making more might cause issues.

Reply:
yea, same reason as above, its not using any specific application and I think other namespaces contain the Application name like `elasticsearch-purevolumestest-0-08-29-20h45m01s`